apps/api/src/applicant/ienapplicant.service.ts

```diff
@@ -155,80 +173,156 @@ export class IENApplicantService {
    * @param applicantUpdate updated fields only status and related field
    * @returns
    */
-  async updateApplicantStatus(
+  async addApplicantStatus(
     id: string,
-    applicantUpdate: IENApplicantUpdateStatusAPIDTO,
+    applicantUpdate: IENApplicantAddStatusAPIDTO,
   ): Promise<IENApplicant | any> {
     const applicant = await this.getApplicantById(id);
-    const { status, status_date, added_by, ha_pcn } = applicantUpdate;
+    const { status, start_date, end_date, added_by, job_id } = applicantUpdate;
     const dataToUpdate: any = {};
     if (added_by) {
       const updated_by_data = await this.ienUsersRepository.findOne(parseInt(added_by));
       if (updated_by_data) {
         dataToUpdate.updated_by = updated_by_data;
       }
     }
+
     const status_obj = await this.ienapplicantUtilService.getStatusById(status);
     dataToUpdate.status = status_obj;
 
-    if (!status_date) {
-      dataToUpdate.status_date = new Date();
-    } else {
-      dataToUpdate.status_date = status_date;
-    }
-    let ha_pcn_obj = null;
-    if (ha_pcn) {
-      ha_pcn_obj = await this.getHaPcn([ha_pcn]);
-      if (ha_pcn_obj.length) {
-        ha_pcn_obj = ha_pcn_obj[0];
-      }
+    dataToUpdate.start_date = null;
+    if (start_date) {
+      dataToUpdate.start_date = start_date;
     }
 
-    await this.ienapplicantRepository.update(applicant.id, {
-      ...dataToUpdate,
-    });
+    dataToUpdate.end_date = null;
+    if (end_date) {
+      dataToUpdate.start_date = end_date;
+    }
 
-    const applicant_obj = await this.getApplicantById(id);
+    let job = null;
```

Comment:
would `let job;` work here?

Reply:
Yes, the job would be null if other than recruitment milestone add/update.